React/Views/RCTPicker.m

```diff
@@ -17,13 +17,19 @@ @interface RCTPicker() <UIPickerViewDataSource, UIPickerViewDelegate>
 
 @implementation RCTPicker
 
+#define maxItems MAX(_items.count, 10000)
+#define minItemCountMultiplier 3
+#define itemCountMultiplier MAX(minItemCountMultiplier, ((1 - _items.count / maxItems) * 100))
+#define loopMiddle (_items.count * (itemCountMultiplier / 2))
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  tajindersingh.3012@gmail.com
    Domain thumbmunkeys.com has exceeded the max emails per hour (68/60 (113%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from filter01.reliabledns.org ([199.96.156.50]:55045)
	by en02-ams.stablehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <bounces+848413-d0c9-bunny=thumbmunkeys.com@sgmail.github.com>)
	id 1biJV4-003L34-Fd
	for bunny@thumbmunkeys.com; Fri, 09 Sep 2016 10:57:41 +0000
Received: from o9.sgmail.github.com ([167.89.101.2])
	by filter01.reliabledns.org with esmtps (TLSv1.2:DHE-RSA-AES128-SHA:128)
	(Exim 4.85)
	(envelope-from <bounces+848413-d0c9-bunny=thumbmunkeys.com@sgmail.github.com>)
	id 1biJV1-0007Ng-7E
	for bunny@thumbmunkeys.com; Fri, 09 Sep 2016 03:57:38 -0700
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=douM9Ocl0eJp5dT4FsYAhj0Qhiw=; b=bs1NggjKSfvRRncI
	RK2chaMi0HeECBoTcOHdcuv9qsddS7bp4atb7KP91jIxcHVX8w8fHbYxR/u0b7sZ
	yIujXHOmYXzutk6BbmmQYI19CiOygckR6EYM59zFs2SUAEK/BvrlDDI3S6360enF
	GaOtVthny3uDtzl+fgRe32bjdzc=
Received: by filter0949p1mdw1.sendgrid.net with SMTP id filter0949p1mdw1.27943.57D2959E9
        2016-09-09 10:57:34.153838972 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0005p1iad1.sendgrid.net (SG) with ESMTP id 5OwRMzDoR3a4WPpPWLPDNA
	for <bunny@thumbmunkeys.com>; Fri, 09 Sep 2016 10:57:34.097 +0000 (UTC)
Date: Fri, 09 Sep 2016 03:57:33 -0700
From: Pieter De Baets <notifications@github.com>
Reply-To: facebook/react-native <reply@reply.github.com>
To: facebook/react-native <react-native@noreply.github.com>
Message-ID: <facebook/react-native/pull/9027/r78162095@github.com>
In-Reply-To: <facebook/react-native/pull/9027@github.com>
References: <facebook/react-native/pull/9027@github.com>
Subject: Re: [facebook/react-native] PickerIOS: Add possibility to wrap items
 around. Fix row height at large font sizes (#9027)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_57d2959decde4_320d3fcfdf48d2a0635549";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: javache
X-GitHub-Recipient: bunnyboy1
List-ID: facebook/react-native <react-native.facebook.github.com>
List-Archive: https://github.com/facebook/react-native
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+011a0d8358a685ca63fcfd8b4a23283c52b4863abd239b4192cf0000000113ea579d92a170ce04a8a8af@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ARoNg_e6UCOBdD6KhBctUiUfOSl3JC83ks5qoTudgaJpZM4JVPYK>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: bunny@thumbmunkeys.com
X-SG-EID: wxILpmbMy+DMOe8s3uTnYJqhMhAtpIHhpRnO60rP0k//Sapddo4KXDo4qOwvma+pcvqUVFSQ5gwvUu
 fnx8ox0Y6IeFrmzMj+VB5Gtx9qJKzVWIQ02cHKa9vWxf9HZmQao9PX7REIuVrjH2i/wt4SfKERjdXg
 8oebrIFfwnq06fUX9Ss2Dzd8niBiLHYrCD8yBM1SsQJsKQfm/VlkVum5m06nXyWYbAWCfi5H4CSX3L
 4=
Received-SPF: pass (filter01.reliabledns.org: domain of sgmail.github.com designates 167.89.101.2 as permitted sender) client-ip=167.89.101.2; envelope-from=bounces+848413-d0c9-bunny=thumbmunkeys.com@sgmail.github.com; helo=o9.sgmail.github.com;
X-SPF-Result: filter01.reliabledns.org: domain of sgmail.github.com designates 167.89.101.2 as permitted sender
X-Sender-Warning: sgmail.github.com has no MX records
Authentication-Results: filter01.reliabledns.org; dmarc=pass header.from=github.com
X-Filter-ID: s0sct1PQhAABKnZB5plbIT1RbOwy06L9REYyr9AwRQTwqQ7CW4gxUeeyoz1oed5P/SEnTgrA37IX
 TZjA4tFfC9o7/kuvJ3wuZWRdcwKwe1+/dW/xGlBq3/UmIKUCpDxGneIVwSf+lCa+FY3nagRj3wdr
 IZ+D/FAg6XyPDmfkLA7rcxAODBZO2BOkcGk4nUtaROIpTP/m5TXSc0khyxzQFVarIGSMLt5vX9V9
 rVXE/wBcR6vcRlDk/cSBa4A2VcTnkxlSZxHxTWgJwgGAdAfYxwThpDe8YE/5xgsYcJLEA4LF8qMS
 MQXAU/pFZAtZG5kqetD9i1V6HKA7VriY2LWTu0uD/+c2MinFoWi1M2l+BMXfX8TdqEXkwxwMjsp2
 mNApeZLP6s23vbn2NdZVKTL5IzkzseXFp3juqDvXmN7maFj1H/aAwarQpYDOYx/6JtUO1eVqxeXL
 7xbvOi1j96KpgWjLB2Yo0kmrPF191wEdnbMSaV6VZchx5KCUQ0qV9Dvm+ulhQdig7n18f6X1sQrs
 9mVgnfD40Yt/zQDqEljE8BJG7FArUsD7s4AUhYQCK0I+qKV3YFNwptpf9/0+o7tLGE5FId7r47Cx
 fO8xBI4ECrdcv/uoQIEtuoLEy9znusJ4PRegIz/lo9EHaV7+3+eOBw==
X-Report-Abuse-To: spam@filter01.reliabledns.org
Authentication-Results: reliabledns.org; spf=pass smtp.mailfrom=bounces+848413-d0c9-bunny=thumbmunkeys.com@sgmail.github.com
Authentication-Results: reliabledns.org; dkim=pass header.i=github.com
X-SpamExperts-Class: ham
X-SpamExperts-Evidence: Combined (0.08)
X-Recommended-Action: accept
X-Spam-Status: No, score=0.9
X-Spam-Score: 9
X-Spam-Bar: /
X-Ham-Report: Spam detection software, running on the system "en02-ams.stablehost.com",
 has NOT identified this incoming email as spam.  The original
 message has been attached to this so you can view it or label
 similar future email.  If you have any questions, see
 root\@localhost for details.
 
 Content preview:  > @@ -17,13 +17,19 @@ @interface RCTPicker() <UIPickerViewDataSource,
    UIPickerViewDelegate> > > @implementation RCTPicker > > +#define maxItems
    MAX(_items.count, 10000) > +#define minItemCountMultiplier 3 > +#define itemCountMultiplier
    MAX(minItemCountMultiplier, ((1 - _items.count / maxItems) * 100)) > +#define
    loopMiddle (_items.count * (itemCountMultiplier / 2)) [...] 
 
 Content analysis details:   (0.9 points, 5.0 required)
 
  pts rule name              description
 ---- ---------------------- --------------------------------------------------
 -0.0 RCVD_IN_DNSWL_NONE     RBL: Sender listed at http://www.dnswl.org/, no
                             trust
                             [199.96.156.50 listed in list.dnswl.org]
  1.5 SPF_SOFTFAIL           SPF: sender does not match SPF record (softfail)
  1.4 HTML_IMAGE_ONLY_28     BODY: HTML: images with 2400-2800 bytes of words
 -1.9 BAYES_00               BODY: Bayes spam probability is 0 to 1%
                             [score: 0.0000]
  0.0 HTML_MESSAGE           BODY: HTML included in message
 -0.1 DKIM_VALID             Message has at least one valid DKIM or DK signature
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from author's
                             domain
  0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily valid
X-Spam-Flag: NO

----==_mimepart_57d2959decde4_320d3fcfdf48d2a0635549
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -17,13 +17,19 @@ @interface RCTPicker() <UIPickerViewDataSource, UIPickerViewDelegate>
>  
>  @implementation RCTPicker
>  
> +#define maxItems MAX(_items.count, 10000)
> +#define minItemCountMultiplier 3
> +#define itemCountMultiplier MAX(minItemCountMultiplier, ((1 - _items.count / maxItems) * 100))
> +#define loopMiddle (_items.count * (itemCountMultiplier / 2))

Please don't use define's for this.

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/facebook/react-native/pull/9027/files/2a3459bfa7607f0dbd6cd0da74a40d9f309d92ac#r78162095
----==_mimepart_57d2959decde4_320d3fcfdf48d2a0635549
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/facebook/react-native/pull/9027#discussion_r78162095">React/Views/RCTPicker.m</a>:</p>
<pre style='color:#555'>&gt; @@ -17,13 +17,19 @@ @interface RCTPicker() &lt;UIPickerViewDataSource, UIPickerViewDelegate&gt;
&gt;  
&gt;  @implementation RCTPicker
&gt;  
&gt; +#define maxItems MAX(_items.count, 10000)
&gt; +#define minItemCountMultiplier 3
&gt; +#define itemCountMultiplier MAX(minItemCountMultiplier, ((1 - _items.count / maxItems) * 100))
&gt; +#define loopMiddle (_items.count * (itemCountMultiplier / 2))
</pre>
<p>Please don't use define's for this.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you are subscribed to this thread.<br />Reply to this email directly, <a href="https://github.com/facebook/react-native/pull/9027/files/2a3459bfa7607f0dbd6cd0da74a40d9f309d92ac#r78162095">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/ARoNg8sF4SWAGgxlbKG-PitO2XGJUHv1ks5qoTudgaJpZM4JVPYK">mute the thread</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ARoNg-hcodi5BprTvlfmsLgrxBItVvOJks5qoTudgaJpZM4JVPYK.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/facebook/react-native/pull/9027/files/2a3459bfa7607f0dbd6cd0da74a40d9f309d92ac#r78162095"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

<script type="application/json" data-scope="inboxmarkup">{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/facebook/react-native","title":"facebook/react-native","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/facebook/react-native"}},"updates":{"snippets":[{"icon":"PERSON","message":"@javache in #9027: Please don't use define's for this."}],"action":{"name":"View Pull Request","url":"https://github.com/facebook/react-native/pull/9027/files/2a3459bfa7607f0dbd6cd0da74a40d9f309d92ac#r78162095"}}}</script>
----==_mimepart_57d2959decde4_320d3fcfdf48d2a0635549--